src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1092,6 +1093,25 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
             ImmutableArray<Location> memberLocations,
             ImmutableArray<bool> memberIsReadOnly);
 
+        /// <summary>
+        /// Classifies a conversion from <paramref name="source"/> to <paramref name="destination"/> according
+        /// to this compilation's programming language.
+        /// </summary>
+        /// <param name="source">Source type of value to be converted</param>
+        /// <param name="destination">Destination type of value to be converted</param>
+        /// <returns>A <see cref="CommonConversion"/> that classifies the conversion from the
+        /// <paramref name="source"/> type to the <paramref name="destination"/> type.</returns>
+        public abstract CommonConversion ClassifyCommonConversion(ITypeSymbol source, ITypeSymbol destination);
+
+        /// <summary>
+        /// Returns true if there is an implicit (C#) or widening (VB) conversion from
+        /// <paramref name="fromType"/> to <paramref name="toType"/>. Returns false if
+        /// either <paramref name="fromType"/> or <paramref name="toType"/> is null, or
+        /// if no such conversion exists.
+        /// </summary>
+        public bool HasImplicitConversion(ITypeSymbol fromType, ITypeSymbol toType)
```

Comment:
We're not super consistent, but the common pattern in Compilation appears to be throwing if parameters are null. Why not here?

Reply:
Because this is what is most useful for clients of this API. I submit that `bool`-returning methods can be resilient against `null` (no conversion exists to or from no type):

``` c#
        public new bool ContainsSyntaxTree(SyntaxTree syntaxTree)
        {
            return syntaxTree != null && _syntaxAndDeclarations.GetLazyState().RootNamespaces.ContainsKey(syntaxTree);
        }
```

Unfortunately, C# and VB do not always agree (e.g. on this API).

Reply:
See https://github.com/dotnet/roslyn/issues/27095

Reply:
It was moved into `Compilation` during the API review. There are no extension methods on `Compilation` in the `CodeAnalysis` assembly, because such methods can just be placed in the API.